lib/modules/platform/github/index.ts

```diff
@@ -166,14 +170,31 @@ export async function initPlatform({
     }
   }
   logger.debug({ platformConfig, renovateUsername }, 'Platform config');
-  const platformResult: PlatformResult = {
+  const githubDS: HostRule[] = [];
+  const gitHubRule: HostRule[] = [
+    {
+      matchHost: 'ghcr.io',
+      username: renovateUsername,
```

Comment:
Is there a reference for this? It's unusual to use a username + token. Also in the hosted app, username is like `renovate[bot]`

Reply:
i think the username will be ignored by github and only the token is used. docker registries normally needs user and passwords, as that's the ways you login via cli too

https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry

Reply:
i didnt find it at start, but then @viceice told me it works with username+token as password.
then i tested it, and turns out he was right.
if u set token only, it doesnt work, 
if u set username+password, it works.

ill look for referenced docs

Reply:
according to these docs:https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry#authenticating-to-the-container-registry
we should use GITHUB_TOKEN but it doesnt work in a rule if i add `token`. i tested already.

on the other hand the username+password work fine.

what is strange is that you are right, the username is Renovate's username not the Username where the packages are saved.

github saves packages on the Owner of the Repository, do we have that?

Reply:
are you sure its renovate[bot]???
it is taking 

```
 await githubApi.getJson<{ login: string; name: string }>(
        endpoint + 'user',
        {
          token,
        }
```

Reply:
so do we need to always call "await getUserDetails" now? just for the host rule?

Reply:
If the username is not important (any valid string works) then just use a dummy string. We only call getUserDetails if `username` is unspecified in config, and it would fail if run with a `ghs_` app token.

Reply:
the hoseted app is explicit setting `username`, so the getUserDetails are not called. Wouldn't work with app token anyways.

Another problem is, the hosted app currently doesn't has access rights for packages, so this will cause new issues on hosted app until it requests the required new rights

Reply:
Hopefully no worse than the 404 it would get today?

Reply:
probably 403 or 401 🤷‍♂️

Reply:
ill try setting user manually and test again just in case

Reply:
Please use "github" to keep it simple, maybe [] aren't permitted now or in future

Reply:
@rarkins what do you mean?
github? where?
[]? should i send undefined instead if it is not filled?

Reply:
@rarkins You approved without resolving this, can you elaborate about this??